### DISCUSSION TITLE
OpenShell Start button spontaneously relocating to center of taskbar

### DISCUSSION THREAD

**Discussion by VulcanTourist** [valinet/ExplorerPatcher#550] [Q&A]
I've been having an issue with the Start Menu button, as altered by Open Shell, spontaneously relocating to the center of the taskbar and then back to the left side when I click around it.  Is this a known issue that will be corrected by the next stable release?

**Comment by valinet** [comment]
I've seen reports of this previously, I don't remember which thread.

Known issue... EP does not mess with taskbar items. 

The button is added by OpenShell. I think this is a problem with that, so maybe open a discussion/issue on their GitHub and see what they know about it/what's the status of it on their part. As far as EP is concerned, I don't know what I could be "fixing" in a subsequent update. Personally, I don't use OpenShell, so I am not really testing with that installed as well.

**Reply by VulcanTourist** [reply]
The thing is, this behavior was not present when I first started to use Windows 11 and ExplorerPatcher.  OpenShell I have been using for years, and it has received no recent updates, but certainly ExplorerPatcher has.  One of those many updates apparently initiated this behavior.  It's only been happening for at most three weeks, likely less.

I frankly don't know that OpenShell is a crucial contributor to this behavior.  It might very well be it would happen even if I wasn't using OpenShell.  All OpenShell does with respect to the taskbar is replace the standard icon with a different one.

**Reply by valinet** [reply]
> One of those many updates apparently initiated this behavior.

This should be backed up by evidence leading to that conclusion, not impressions. What investigations have you taken to conclude that?

> All OpenShell does with respect to the taskbar is replace the standard icon with a different one.

Have you actually studied how it works beyond the visual part? It’s not replacing any icon, if it did, the thing wouldn’t move as you said.

> It might very well be it would happen even if I wasn't using OpenShell. 

Another statement that would better be backed by some actual evidence leading to its claims.

EP is in a public repository with public source code, everyone is encouraged to investigate any issue they may be facing, considering there are more than enough instruments available to do so, and eventually maybe contribute back the results of their findings and an eventual patch. It’s hard for a niche potential issue to ever be “fixed”, if really an issue, without the reporter documenting it a bit more, doing some tests, determining when it started, on what commit, what might cause it, and even a sketch for a solution.

Known issues are generally represented by open issues in the **Issues** section of the repository. Those, some information from discussions, the wiki and the actual source code are everything that’s “known” in regard to this project. I don’t have a separate backlog with tasks to do and secret stuff I know about or work on, the development is done all in the open, one of the hopes being that the community will also step up and contribute from time to time.

**Comment by sisericm** [comment]
Which versions of EP and Openshell are you using @VulcanTourist ?
FWIW I am currently on the following and don't encounter this behaviour.
![Image 001](https://user-images.githubusercontent.com/8255994/145720224-20a51013-33dd-4dcd-9f20-c82245a0500f.png)

![Image 002](https://user-images.githubusercontent.com/8255994/145720260-f50d9bc5-355b-4a87-bee7-2b3fc03cb5d7.png)

**Reply by VulcanTourist** [reply]
I'm using the same, but since the primary author is skeptical or indifferent, I lack any applicable skill, and the problem occurs only sporadically triggered by some event I haven't noticed, I'm just going to live with it.  Maybe I'll later notice the triggering event and be able to pursue it further.

**Reply by valinet** [reply]
@VulcanTourist I am not skeptical or indifferent, if I were, I wouldn't even have bothered to respond. I never understand this attitude. What do you expect from me? Do you think I am some kind of wizard that can magically guess what the problem is and fix it? I am open to "fixing" it, but how? It's an issue happening on your side; you haven't even determined what it is caused by, let alone attempting to repair/workaround it. Not to mention that I don't understand why it's expected now for this project to also fix third party apps, besides Microsoft's mess. And as I said, there are threads here and there of people reporting similar behavior. I don't think it's an EP issue at all. Do you see any code in the program that moves OpenShell's Start button...?

Anyway, Google search, maybe try what they mention here, maybe it helps:

https://www.elevenforum.com/t/open-shell-4-4-170-moves-its-icon-solved.1852/

Please report back if it helps in any way. Thanks.

P.S. Also, a mention in general. This is not a Windows 11 forum, or a forum in general. As many have noticed, most of the replies on the threads over here are given by me, and I do not always posses the best advice or knowledge at all in some areas. So it's not that I suggest not posting here, but maybe there are sometimes bigger chances of success with making a post here, for example, and also linking it in a post on [elevenforum](https://www.elevenforum.com), for example, where the engagement by third party users seems to be a big more enthusiastic than over here. Furthermore, a lot of "problems" a lot of the times do not have anything to do with EP. Not to mention the people that keep opening issues, even when it seems like an issue, despite the rule stating clearly not to. For something to be open in issues, it has to be discussed on and confirmed as an issue, via a thread in Discussions.

**Reply by VulcanTourist** [reply]
> Not to mention that I don't understand why it's expected now for this project to also fix third party apps, besides Microsoft's mess. And as I said, there are threads here and there of people reporting similar behavior. I don't think it's an EP issue at all. Do you see any code in the program that moves OpenShell's Start button...?

It was my understanding that one of EP's roles was to reposition the items on the taskbar, including the start button, from the center of the taskbar back to being left-justified.  There's also an option to reposition the start menu.  It seemed likely to me that this spontaneous un-re-positioning of the button was an issue with that feature.  I didn't anticipate it being an issue with Open Shell, since I would still have a start button in need of being repositioned even if I configured Open Shell not to replace the button with its own icon (which I have now done).

I reported what I knew of the behavior, which wasn't very much, partly in the hope that it was familiar to someone else who might know more.  The only tool I have to understand it better is an eagle-eye focus on what happened just before it jumped that might have been the cause, which is damned hard when your focus was on whatever else you were trying to accomplish at the time.  Usually I was noticing the jump some time later after it had happened.  I thought the tone of your initial reply and this one suggested something like skepticism or defensiveness.  That is how I perceived it.

**Comment by Galixte** [comment]
@sisericm maybe you could begin by comparing with previous versions than 22000.348.40.6 to find which revision of ExplorerPatcher reveals a possible issue?

**Reply by kichrot** [reply]
I had this effect until I updated WINDOWS to 22000.348. After the update, there has never been.

**Reply by VulcanTourist** [reply]
> I had this effect until I updated WINDOWS to 22000.348. After the update, there has never been.

I'm using 22000.318 and don't have any option to update it further.  I changed Open Shell to not replace the default button.  I haven't noticed it relocate the button since then.

**Reply by valinet** [reply]
You can also upgrade to 22000.376 using Windows Update. It's out now for all Windows 11 users.

**Reply by valinet** [reply]
> > I had this effect until I updated WINDOWS to 22000.348. After the update, there has never been.
> 
> I'm using 22000.318 and don't have any option to update it further. I changed Open Shell to not replace the default button. I haven't noticed it relocate the button since then.

OpenShell relocated its button. It draws a custom buttons it skins and what not. It odes not take control of the original Start button from the original taskbar.
